python/dgl/data/movielens.py

```diff
@@ -0,0 +1,658 @@
+
+    Reference:
+    [1] Bradley N Miller, Istvan Albert, Shyong K Lam, Joseph A Konstan, and John Riedl. 
+    Movielens unplugged: experiences with an occasionally connected recommender system. In Proceedings of
+    the 8th international conference on Intelligent user interfaces, pp. 263-266. ACM, 2003.
+
+    Statistics:
+
+    MovieLens-100K (ml-100k)
+
+    - Users: 943
+    - Movies: 1,682
+    - Ratings: 100,000 (1, 2, 3, 4, 5)
+
+    MovieLens-1M (ml-1m)
+
+    - Users: 6,040
+    - Movies: 3,706
+    - Ratings: 1,000,209 (1, 2, 3, 4, 5)
+
+    MovieLens-10M (ml-10m)
+
+    - Users: 69,878 
+    - Movies: 10,677
+    - Ratings: 10,000,054 (0.5, 1, 1.5, ..., 4.5, 5.0)
+
+    Parameters
+    ----------
+    name: str
+        Dataset name. (:obj:`"ml-100k"`, :obj:`"ml-1m"`, :obj:`"ml-10m"`). 
+    valid_ratio: int
+        Ratio of validation samples out of the whole dataset. Should be in (0.0, 1.0).
+    test_ratio: int
+        Ratio of testing samples out of the whole dataset. Should be in (0.0, 1.0). This parameter is invalid
+        when :obj:`name` is :obj:`"ml-100k"`, since testing samples are pre-specfied for :obj:`"ml-100k"`.
+        Default: None
+    text_cache: str
+        Raw file directory to store the pre-trained word embeddings that used to preprocess movie titles.
+        Default: ~/.dgl/movielens_vector_cache/
+    raw_dir : str
+        Raw file directory to download/contain the input data directory.
+        Default: ~/.dgl/
+    force_reload : bool
+        Whether to reload the dataset. Default: False
+    verbose : bool
+        Whether to print out progress information. Default: True.
+    reverse_edge : bool
+        Whether to add reverse edges in graph. Default: True.
+    transform : callable, optional
+        A transform that takes in a :class:`~dgl.DGLGraph` object and returns
+        a transformed version. The :class:`~dgl.DGLGraph` object will be
+        transformed before every access.
+
+    Attributes
+    ----------
+    train_graph: dgl.DGLGraph
+        Training graph
+    valid_graph: dgl.DGLGraph
+        Validation graph
+    test_graph: dgl.DGLGraph
+        Testing graph
+    info: dict
+        Store training, validation and testing rating pairs.
+    feat: dict
+        Store input features of users and movies
+
+    Notes
+    -----
+    - When using MovieLens, users are required to install :obj:`torchtext>=0.15.1`, which provides :obj:`Glove` API to preprocess text embeddings.
+    - The number of edges is doubled to form an undirected(bidirected) graph structure.
+    - Each time when the dataset is loaded with a different setting of training, validation and testing ratio, the param
+    :obj:`"force_reload"` should be set to :obj:`"True"`. Otherwise the previous split of dataset will be loaded and not be
+    updated in the storage.
+
+    Examples
+    --------
+    >>> dataset = MovieLens()
```

Comment:
This should be `MovieLensDataset`. Add code for importing the class. Make sure that users can run this example without errors.

Reply:
fix in the latest commit